### ISSUE TITLE
Not all parameters in YOLO-World pretrained weights can be loaded

### ISSUE THREAD

**Issue by chongkuiqi** [ultralytics/ultralytics#18688]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

When i init a YOLOv8s-World-v2 model from a .yaml configuration file and load pretrained weights, it seems that three parameters can not be loaded.  The code and outputs are as follows:
```python
from ultralytics import YOLOWorld
model_path = './yolov8s-worldv2.pt'
model_cfg = "./yolov8s-worldv2.yaml"
model = YOLOWorld(model_cfg).load(model_path)

# outputs
Transferred 409/412 items from pretrained weights
```

The yolov8s-worldv2.yaml configuration file is copied from `ultralytics/ultralytics/cfg/models/v8/yolov8-worldv2.yaml` without any modifications except for the file name.

And i find a solution, in `ultralytics/ultralytics/nn/modules/block.py`, modify the `class BNContrastiveHead(nn.Module)` as follows:
```python
# self.bias = nn.Parameter(torch.tensor([-10.0]))
self.bias = nn.Parameter(-10.0 * torch.ones([]))
```
Rerun the above code and output as:
```python
Transferred 412/412 items from pretrained weights
```

### Environment

```
Ultralytics 8.3.59 🚀 Python-3.10.14 torch-2.2.2+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24259MiB)
Setup complete ✅ (32 CPUs, 62.8 GB RAM, 686.0/915.3 GB disk)

OS                  Linux-5.15.0-130-generic-x86_64-with-glibc2.31
Environment         Linux
Python              3.10.14
Install             git
RAM                 62.79 GB
Disk                686.0/915.3 GB
CPU                 Hygon C86 3285  8-core Processor
CPU count           32
GPU                 NVIDIA GeForce RTX 3090, 24259MiB
GPU count           1
CUDA                12.1

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.2.2+cu121>=1.8.0
torch               ✅ 2.2.2+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.17.2+cu121>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0
```

### Minimal Reproducible Example

```python
from ultralytics import YOLOWorld
model_path = './yolov8s-worldv2.pt'
model_cfg = "./yolov8s-worldv2.yaml"
model = YOLOWorld(model_cfg).load(model_path)

# outputs
Transferred 409/412 items from pretrained weights
```
The yolov8s-worldv2.yaml configuration file :
```
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLOv8-World-v2 object detection model with P3-P5 outputs. For details see https://docs.ultralytics.com/tasks/detect

# Parameters
nc: 80 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024] # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
  s: [0.33, 0.50, 1024] # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
  m: [0.67, 0.75, 768] # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
  l: [1.00, 1.00, 512] # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
  x: [1.00, 1.25, 512] # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]] # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, C2fAttn, [512, 256, 8]] # 12

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, C2fAttn, [256, 128, 4]] # 15 (P3/8-small)

  - [15, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]] # cat head P4
  - [-1, 3, C2fAttn, [512, 256, 8]] # 18 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]] # cat head P5
  - [-1, 3, C2fAttn, [1024, 512, 16]] # 21 (P5/32-large)

  - [[15, 18, 21], 1, WorldDetect, [nc, 512, True]] # Detect(P3, P4, P5)

```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @chongkuiqi, thank you for bringing this to our attention 🚀! We recommend checking the [Ultralytics Documentation](https://docs.ultralytics.com) to ensure you're familiar with our [Models](https://docs.ultralytics.com/models/), [Tasks](https://docs.ultralytics.com/tasks/), and helpful [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For 🐛 **Bug Reports**, we kindly request a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) that reproduces the issue you mentioned. It looks like you've already provided some context, which is great! However, confirming that this issue persists after upgrading to the latest version would be very helpful.

### Troubleshooting Steps

Please ensure you're using the latest `ultralytics` package by upgrading all requirements in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Verified Environments

YOLO models can be run in any of the following up-to-date verified environments. You may want to test within one of these platforms to verify the issue persists:
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community and Support

If this is related to a custom training ❓ question or more detailed usage issue, you can join our thriving Ultralytics community! Engage in real-time discussions on [Discord](https://discord.com/invite/ultralytics) 🎧, share insights on [Discourse](https://community.ultralytics.com), or exchange ideas on our [Subreddit](https://reddit.com/r/Ultralytics). 

### Status and Build Integrity

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

The current status of [Continuous Integration (CI)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml) indicates whether all test cases are passing across macOS, Windows, and Ubuntu systems. We highly recommend ensuring your system matches the dependencies outlined in the repository and documentation.

This is an automated response to help address your issue faster 😊. An Ultralytics engineer will respond shortly to provide further assistance!

**Comment by Y-T-G** [comment]
You can open a PR for the fix
